src/coreclr/jit/assertionprop.cpp

```diff
@@ -3416,7 +3443,14 @@ bool Compiler::optZeroObjAssertionProp(GenTree* tree, ASSERT_VALARG_TP assertion
         return false;
     }
 
-    unsigned       lclNum         = tree->AsLclVar()->GetLclNum();
+    // No ZEROOBJ assertions for simd.
+    //
+    if (varTypeIsSIMD(tree))
```

Comment:
Do you have an example of diff from this change?

Reply:
This is here to prevent an assert, since we're generating (and now more thoroughly propagating) ZEROBJ assertions for SIMD trees, and code elsewhere is not prepared to handle it.